src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/cif/file/internal/CifFileReader.java

```diff
@@ -0,0 +1,53 @@
+package pl.edu.icm.pl.mxrdr.extension.importer.cif.file.internal;
+
+import edu.harvard.iq.dataverse.importer.metadata.ImporterFieldKey;
+import org.rcsb.cif.CifIO;
+import org.rcsb.cif.model.CifFile;
+import org.rcsb.cif.schema.StandardSchemata;
+import org.rcsb.cif.schema.mm.MmCifFile;
+import pl.edu.icm.pl.mxrdr.extension.importer.cif.file.CifFileFormKeys;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class CifFileReader {
+    private static final Logger logger = Logger.getLogger(CifFileReader.class.getSimpleName());
+
+    private static final String CIF_URL_TEMPLATE = "https://files.rcsb.org/header/%s.cif";
+
+    // -------------------- LOGIC --------------------
+
+    public static Optional<MmCifFile> readAndParseCifFile(Map<ImporterFieldKey, Object> importerInput) {
```

Comment:
This method should accept `String` and `File` parameters directly instead of a map.

Reply:
Also `Optional` seems over-defensive, as there cannot be `null` returned from `CifIO`.